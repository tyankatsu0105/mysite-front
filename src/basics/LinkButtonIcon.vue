<template>
	<a
		:href="href"
		target="_blank"
		class="LinkButtonIcon"
	>
		<font-awesome-icon :icon="icon" />
	</a>
</template>

<script>
export default {
  props: {
    icon: {
      type: Object,
      requirer: false
    },
    href: {
      type: String,
      requirer: true
    }
  }
};
</script>

<style scoped lang='scss'>
.LinkButtonIcon {
  position: relative;
  padding: 8px 24px;
  font-size: 1.3rem;
  background-color: $color-black;
  border-radius: 8px;
  box-shadow: 0 2px 5px 0 #1e1e1e;
  transition: $tst-default;

  &:hover {
    box-shadow: 0 2px 6px #0e0e0e;
    transform: translateY(-2px);
  }
}
</style>
