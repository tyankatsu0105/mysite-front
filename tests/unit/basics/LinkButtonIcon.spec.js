import { shallowMount } from "@vue/test-utils";
import LinkButtonIcon from "@/basics/LinkButtonIcon.vue";

describe("LinkButtonIcon", () => {
  it("render", () => {
    const wrapper = shallowMount(LinkButtonIcon);
    wrapper.setProps({
      icon: {
        prefix: "fab",
        iconName: "twitter"
      },
      href: "hogehoge.com"
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
