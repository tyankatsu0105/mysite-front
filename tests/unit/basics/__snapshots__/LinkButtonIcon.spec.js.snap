// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButtonIcon render 1`] = `
<a target="_blank" class="LinkButtonIcon" href="hogehoge.com">
  <font-awesome-icon-stub icon="[object Object]"></font-awesome-icon-stub>
</a>
`;
